### PR TITLE
Fix Tests with multiple threads.

### PR DESCRIPTION
**Pull request by hamhut1066**
The tests have been fixed to sort the jobs correctly (based on run time).

### CODE REVIEW

File: spec/tom_queue/tom_queue_integration_spec.rb [hunk 1]
```diff
@@ -212,7 +212,7 @@ def start!
 
       # Merge the list of received work together sorted by received time,
       # and compare to the source list
-      sink_order = consumers.map { |c| c.work }.flatten.sort.map { |a| a.payload }
+      sink_order = consumers.map(&:work).flatten.sort { |a, b| a[:run_at] <=> b[:run_at] }.map(&:payload)
```
Comment (hamhut1066) [inline spec/tom_queue/tom_queue_integration_spec.rb]: Have I sorted this against the wrong time? `run_at` instead of `received_at`?

Comment (afcapel) [inline spec/tom_queue/tom_queue_integration_spec.rb]: You can also use `Enumerable#sort_by`.

http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-sort_by

Comment (joshado) [inline spec/tom_queue/tom_queue_integration_spec.rb]: Sorting `run_at` will sort based on when the job is scheduled to run. `run_at` is the header on the message indicating the original value of `run_at` passed into the publish method, so no longer actually compares the order that they were processed.

So, no, probably best not to use that as a comparison ;)

File: spec/tom_queue/tom_queue_integration_spec.rb [hunk 2]
```diff
@@ -253,7 +253,7 @@ def start!
 
       # Now merge all the consumers internal work arrays into one
       # sorted by the received_at timestamps
-      sink_order = consumers.map { |c| c.work }.flatten.sort.map { |a| a.payload }
+      sink_order = consumers.map(&:work).flatten.sort { |a, b| a[:run_at] <=> b[:run_at] }.map(&:payload)
```
Comment (joshado) [inline spec/tom_queue/tom_queue_integration_spec.rb]: See above, breaks the tests in such a way that they reliably pass, but don't actually test anything ;)

**Pull request closed without merging**